packages/nimble-components/src/select/index.ts

```diff
@@ -60,36 +196,205 @@ export class Select extends FoundationSelect implements ErrorPattern {
+        );
+
+        if (!captured?.disabled) {
+            this.updateSelectedIndexFromFilteredSet();
         }
+        return super.clickHandler(e);
     }
 
-    private controlChanged(
-        _prev: HTMLElement | undefined,
-        _next: HTMLElement | undefined
+    public inputClickHandler(e: MouseEvent): void {
+        e.stopPropagation(); // clicking in filter input shouldn't close dropdown
+    }
+
+    /**
+     * Handle content changes on the control input.
+     *
+     * @param e - the input event
+     * @internal
+     */
+    public inputHandler(e: InputEvent): boolean {
+        this.filter = this.filterInputElement?.value ?? '';
+        if (!this.committedSelectedOption) {
+            this.committedSelectedOption = this._options.find(
+                option => option.selected
+            );
+        }
+        this.clearSelection();
+        this.filterOptions();
+
+        if (
+            this.filteredOptions.length > 0
+            && this.committedSelectedOption
+            && !this.filteredOptions.includes(this.committedSelectedOption)
+        ) {
+            this.committedSelectedOption.selected = false;
+            const enabledOptions = this.filteredOptions.filter(
+                o => !o.disabled
+            );
+            if (enabledOptions.length > 0) {
+                enabledOptions[0]!.selected = true;
+            } else {
+                // only filtered option is disabled
+                this.selectedOptions = [];
+                this.selectedIndex = -1;
+                this.filteredOptions[0]!.selected = false;
+            }
+        } else if (this.committedSelectedOption) {
+            this.committedSelectedOption.selected = true;
+        }
+
+        if (e.inputType.includes('deleteContent') || !this.filter.length) {
+            return true;
+        }
+
+        e.stopPropagation();
+        return true;
+    }
+
+    // disabling linting since the override return type should match the parent class return type
+    // eslint-disable-next-line @typescript-eslint/no-invalid-void-type
+    public override keydownHandler(e: KeyboardEvent): boolean | void {
+        const key = e.key;
+        if (e.ctrlKey || e.shiftKey) {
+            return true;
+        }
+
+        switch (key) {
+            case keySpace: {
+                if (this.filterMode === FilterMode.none) {
+                    return super.keydownHandler(e);
+                }
+                // when dropdown is open allow user to enter a space for filter text
+                // (calling super method will close dropdown)
+                if (!this.open) {
+                    super.keydownHandler(e);
+                }
+                break;
+            }
+
+            case keyEnter: {
+                this.updateSelectedIndexFromFilteredSet();
+                super.keydownHandler(e);
+                this.focus();
+                break;
+            }
+            case keyEscape: {
```

Comment:
I haven't reviewed the most recent updates to this PR, but something still seems off with the Escape behavior.

Specifically, I see this:
1. Open the storybook page for the select with filterMode = standard
2. Type a filter that matches something (e.g. "Zur")
3. See that the matching option is highlighted in the dropdown
4. Press Escape
5. Notice that the dropdown closes and reverts back to the original value
6. Click on the select to open it again
7. Notice that the current value is not highlighted in the dropdown and that filtering doesn't cause anything to become highlighted

Reply:
Yep. Missed a needed subtle change. I've updated and added a test.